### ISSUE TITLE
`focus` is a concept that only applies to `window` and `input-area`

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2464]
With respect to buffers, it makes little sense to say which one has the focus.

There are buffers and, at any moment, there's a single `current-buffer` (which may not correspond to the displayed one).

To stress the fact that buffers can also be switched to, i.e. bring them to display, it would be better to use concepts such as background/foreground.

I suggest switching from:

```
(define-ffi-generic ffi-window-set-buffer (window buffer &key focus))
```

to:

```
(define-ffi-generic ffi-window-set-buffer (window buffer &key foreground))
```

If you're wondering about `set-current-buffer`, 
```
(defun set-current-buffer (buffer &key (focus t)) ...)
```

, there are only three of the following calls `(set-current-buffer buffer :focus nil)`, which should probably be replaced by `(with-current-buffer ...)`. 

In other words, I'm saying that the `focus` key of `set-current-buffer` should be deprecated.

**Comment by Ambrevar** [comment]
Note however that `:focus nil` is actually a GTK-implementation detail.  It's not about displaying the buffer in the foreground or not, but whether the widget is "grabbed".  So yes, focus applies to buffer in the sense that if they have a selected HTML input, then this is where the user is going to type.

In `buffer-source`, we have

```lisp
   (prompter:selection-actions #'(lambda (buffer)
                                   (set-current-buffer buffer :focus nil)))
```

which is meant to effective display the buffer to the foreground, but without focusing the GTK widget so that the user keeps typing in the prompt-buffer.

All that said, the naming and the API is confusing.  It could be improved, but `with-current-buffer` won't work here because it does not display the buffer.

**Comment by lansingthomas** [comment]
This came up in my ongoing exploration of rethinking the prompt area UI (#2687) and I think it belongs here.

![Issue PA 1 - inconsistent buffer focus selectors](https://user-images.githubusercontent.com/6666079/211422048-c6155e79-d322-4df2-8486-20d6e06049b0.png)

I see this as an issue because the interaction:
1. is not consistent with the expectation users bring from other apps
2. does not have a visible status
3. causes errors in other places when users type things in random websites intending to type in the Prompter of our Prompt Area

Make sense? (I tried to explain this in the recently closed #2706, but I was not really getting the issue yet.)

**Comment by aartaka** [comment]
Yes, makes total sense!

**Comment by aadcg** [comment]
@lansingthomas I've moved your valuable comment [here (see Problem 3)](https://github.com/atlas-engineer/nyxt/issues/2687#issue-1472908764).

The purpose of this issue is slightly different so, with your authorization, I'd like to remove your comment from here so that we keep discussions focused.

**Comment by aadcg** [comment]
Superseded by #3470.
